open_dread_rando/dread_patcher.py

```diff
@@ -98,10 +99,9 @@ def chunks(l, n):
     return lua_util.replace_lua_template("custom_init.lua", replacement)
 
 
-def patch_pickups(editor: PatcherEditor, lua_scripts: LuaEditor, pickups_config: list[dict], configuration: dict):
+def patch_pickups(editor: PatcherEditor, map_icon_editor: MapIconEditor, lua_scripts: LuaEditor, pickups_config: list[dict], configuration: dict):
```

Comment:
The MapIconEditor is used in 3 places now: 

- in pickup patcher where it used to be used
- in `static_fixes` where it fixes the icon defs (bmmdef) to use the same icon for each side with different offsets, as well as goes through all the map files to reverse existing shields
- in `dread_patcher.patch_doors()` which calls the function to add custom door rando assets

Since it's used by multiple APIs after these changes, I decided to treat it like a `PatcherEditor` where the main function instantiates it and passes it to the APIs that require it. It might be better to make `PatcherEditor` create and hold a reference to the instance of it now that I think about it.

Reply:
> It might be better to make `PatcherEditor` create and hold a reference to the instance of it now that I think about it.

agreed